File: app/src/main/java/org/openstaxcollege/android/adapters/BookmarkRecyclerViewAdapter.java
/**
 * Copyright (c) 2015 Rice University
 *
 * This software is subject to the provisions of the GNU Lesser General
 * Public License Version 2.1 (LGPL).  See LICENSE.txt for details.
 */
package org.openstaxcollege.android.adapters;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import org.openstaxcollege.android.R;
import org.openstaxcollege.android.activity.WebViewActivity;
import org.openstaxcollege.android.beans.Content;
import org.openstaxcollege.android.providers.Bookmarks;
import org.openstaxcollege.android.utils.OSCUtil;

import java.util.ArrayList;

import co.paulburke.android.itemtouchhelperdemo.helper.ItemTouchHelperAdapter;

/** Adapter to properly display bookmarks in RecyclerView
 * @author Ed Woodward
 * */
public class BookmarkRecyclerViewAdapter extends RecyclerView.Adapter<BookmarkRecyclerViewAdapter.ViewHolder> implements ItemTouchHelperAdapter
{
    /** List of Content objects to display*/
    private ArrayList<Content> contentList;
    Content content;
    static Context context;

    private int rowLayout;

    public BookmarkRecyclerViewAdapter(ArrayList<Content> content, int rowLayout, Context context)
    {
        contentList = content;
        this.rowLayout = rowLayout;
        this.context = context;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i)
    {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(rowLayout, viewGroup, false);
        return new ViewHolder(v,contentList);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int i)
    {
        content = contentList.get(i);
        viewHolder.title.setText(content.getTitle());
        viewHolder.other.setText(content.getContentString());
        if (viewHolder.logo != null && content.getIcon() != null)
        {
            viewHolder.logo.setImageResource(OSCUtil.getCoverId(content.getIcon(), context));

        }

    }

    @Override
    public int getItemCount()
    {
        return contentList == null ? 0 : contentList.size();
    }

    @Override
    public void onItemDismiss(int position)
    {
        Content currentContent = contentList.get(position);
        context.getContentResolver().delete(Bookmarks.CONTENT_URI, "_id="+ currentContent.getId(), null);
        contentList.remove(position);
        notifyItemRemoved(position);
        Toast.makeText(context, "Bookmark deleted for " + currentContent.getTitle(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public boolean onItemMove(int fromPosition, int toPosition)
    {
        return true;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener
    {
        public ImageView logo;
        public TextView title;
        public TextView other;
        public View view;
        ArrayList<Content> contentList;

        public ViewHolder(View itemView, ArrayList<Content> contentList)
        {
            super(itemView);
            view = itemView;
            this.contentList = contentList;

            logo = (ImageView) itemView.findViewById(R.id.logoView);
            title = (TextView)itemView.findViewById(R.id.bookName);
            other = (TextView)itemView.findViewById(R.id.other);
            itemView.setOnClickListener(this);
        }


        @Override
        public void onClick(View v)
        {
            Content content = contentList.get(getAdapterPosition());
            Log.d("BRVA", "title: " + content.getBookTitle());
            Content bookTitle = OSCUtil.getTitle(content.getBookTitle(), context);
            if(bookTitle != null)
            {
                content.setBookUrl(bookTitle.getBookUrl());
            }
            Context context = v.getContext();
            Intent wv = new Intent(v.getContext(), WebViewActivity.class);
            wv.putExtra(v.getContext().getString(R.string.webcontent), content);

            context.startActivity(wv);
        }


    }
}

File: app/src/main/java/org/openstaxcollege/android/adapters/BookmarkRecyclerViewAdapter.kt
/**
 * Copyright (c) 2015 Rice University
 *
 * This software is subject to the provisions of the GNU Lesser General
 * Public License Version 2.1 (LGPL).  See LICENSE.txt for details.
 */
package org.openstaxcollege.android.adapters

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

import org.openstaxcollege.android.R
import org.openstaxcollege.android.activity.WebViewActivity
import org.openstaxcollege.android.beans.Content
import org.openstaxcollege.android.providers.Bookmarks
import org.openstaxcollege.android.utils.OSCUtil

import java.util.ArrayList

import co.paulburke.android.itemtouchhelperdemo.helper.ItemTouchHelperAdapter

/** Adapter to properly display bookmarks in RecyclerView
 * @author Ed Woodward
 */
class BookmarkRecyclerViewAdapter(
        /** List of Content objects to display */
        private val contentList: ArrayList<Content>?, private val rowLayout: Int, private val context: Context) : RecyclerView.Adapter<BookmarkRecyclerViewAdapter.ViewHolder>(), ItemTouchHelperAdapter
{

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder
    {
        val v = LayoutInflater.from(viewGroup.context).inflate(rowLayout, viewGroup, false)
        return ViewHolder(v, contentList!!)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int)
    {
        val content = contentList!![i]
        viewHolder.title.text = content.title
        viewHolder.other.text = content.contentString
        if(viewHolder.logo != null && content.icon != null)
        {
            viewHolder.logo.setImageResource(OSCUtil.getCoverId(content.icon!!, context))

        }

    }

    override fun getItemCount(): Int
    {
        return contentList?.size ?: 0
    }

    override fun onItemDismiss(position: Int)
    {
        //Log.d("BRVA","position = " + position);
        //Log.d("BRVA","content list size = " + contentList!!.size);
        if(position < contentList!!.size)
        {
            val currentContent = contentList[position]
            context.contentResolver.delete(Bookmarks.CONTENT_URI, "_id=" + currentContent.id, null)
            contentList.removeAt(position)
            notifyItemRemoved(position)
            Toast.makeText(context, context.getString(R.string.bookmark_deleted) + currentContent.title!!, Toast.LENGTH_SHORT).show()
        }
    }

    override fun onItemMove(fromPosition: Int, toPosition: Int): Boolean
    {
        return true
    }

    inner class ViewHolder constructor(var view: View, internal var contentList: ArrayList<Content>) : RecyclerView.ViewHolder(view), View.OnClickListener
    {
        val logo: ImageView
        var title: TextView
        var other: TextView

        init
        {

            logo = view.findViewById<View>(R.id.logoView) as ImageView
            title = view.findViewById<View>(R.id.bookName) as TextView
            other = view.findViewById<View>(R.id.other) as TextView
            view.setOnClickListener(this)
        }


        override fun onClick(v: View)
        {
            val content = contentList[adapterPosition]
            //Log.d("BRVA", "title: " + content.getBookTitle());
            val bookTitle = OSCUtil.getTitle(content.bookTitle, v.context)
            if(bookTitle != null)
            {
                content.bookUrl = bookTitle.bookUrl
            }
            val context = v.context
            val wv = Intent(v.context, WebViewActivity::class.java)
            wv.putExtra(v.context.getString(R.string.webcontent), content)

            context.startActivity(wv)
        }


    }
}
